### PR TITLE
feat: direct access to gated resources if user is authorized

### LINKED ISSUE #1104: Give access to gated resources if user is logged in https://support.moleculardevices.com/
If user is already logged in the support portal i.e. https://support.moleculardevices.com/ then a cookie will be created and on the basis of that value in moleculardevices.com, all gated resources will appear as ungated to that particular user.

Example cookie value: 
image

Here are the all possible values of cookie

DISTRIBUTOR
INTEGRATOR
SALES
TECH

**Regards,
Rajneesh**

### PR DESCRIPTION
**Pull request by duynguyen**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #1104 
To be an authorized user, set cookie `STYXKEY_PortalUserRole` to one of these values:
DISTRIBUTOR
INTEGRATOR
SALES
TECH

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-i3x-readers#resources
- After: https://1104--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-i3x-readers#resources

### CODE REVIEW

File: blocks/card/card.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ class Card {
       ? item.imageBlock : createOptimizedPicture(itemImage, item.title, 'lazy', [{ width: '800' }]);
 
     let cardLink = item.path;
-    if (item.gated === 'Yes' && item.gatedURL && item.gatedURL !== '0') {
+    if (isGatedResource(item) && item.gatedURL && item.gatedURL !== '0') {
```
Comment (alexcarol) [inline blocks/card/card.js:87]: Why doesn't this function include `item.gatedURL && item.gatedURL !== '0'`? It looks like that's a requirement for resources to be considered gated too.

Comment (duynguyen) [reply blocks/card/card.js]: good point, thank you! fixed.

**Comment by lydiapuric** [comment]
Hmm, tried it on https://1104--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-i3x-readers#resources, cleared all cookies, not setting of STYXKEY_PortalUserRole cookie
I still can read the full application note...

**Comment by duynguyen** [comment]
@lydiapuric did you see some app notes that are gated but having direct link on this page? can you share the app note name so I can check?

**Comment by lydiapuric** [comment]
@duynguyen NanoBRET: https://1104--moleculardevices--hlxsites.hlx.live/en/assets/app-note/br/nanobret-technology-on-spectramax-i3x-multi-mode-microplate-readers

**Comment by lydiapuric** [comment]
NanoBRET is not gated, I will recheck with a gated one. Sorry for the noise :(

**Comment by lydiapuric** [comment]
@duynguyen , when I know the gated url, I can access that page without having the cookie checked. IMO we need to prohibit this resource in that case too

**Comment by duynguyen** [comment]
@lydiapuric yes that's correct - anyone knowing the file URL can access it directly without being "authorized" (which is soft anyway as you can simply set the cookie client side).

I'm not sure if we should do it differently though, as that's how the customer wants it in the old site as well. tagging @dev-rajneeshkumar to confirm.

**Comment by mhaack** [comment]
@lydiapuric @duynguyen I think this was the same before. The entire gated url concept is soft only. If you know the doc url you can always bypass the landing page.

**Comment by lydiapuric** [comment]
@duynguyen No, didn't find a gated App Note

**Review by lydiapuric — APPROVED** [review]
LGTM +1
Soft access is addressed and works.

**Comment by lydiapuric** [comment]
@mhaack and @duynguyen Just checked https://live-md30.pantheonsite.io/sites/default/files/resources/tagging/elisa.pdf and you are right, the gated resource is also here accessible with the direct URL.

**Review by alexcarol — APPROVED** [review]

**Review by mhaack — APPROVED** [review]

**Pull request merged**